dhall-docs/src/Dhall/Docs.hs

```diff
@@ -106,20 +119,32 @@ getAllDhallFilesAndHeaders
     :: Path Abs Dir -- ^ Base directory to do the search
     -> IO [(Path Abs File, Header)]
 getAllDhallFilesAndHeaders baseDir = do
-    files <- snd <$> Path.IO.listDirRecur baseDir
+    files <- filter hasDhallExtension . snd <$> Path.IO.listDirRecur baseDir
     Data.Maybe.catMaybes <$> mapM readDhall files
   where
+    hasDhallExtension :: Path Abs File -> Bool
+    hasDhallExtension absFile = case Path.splitExtension absFile of
+        Nothing -> False
+        Just (_, ext) -> ext == ".dhall"
+
     readDhall :: Path Abs File -> IO (Maybe (Path Abs File, Header))
     readDhall absFile = do
         let filePath = Path.fromAbsFile absFile
-        fileContents <- Data.ByteString.readFile filePath
-        return $ case Text.Encoding.decodeUtf8' fileContents of
-            Left _ -> Nothing
-            Right contents ->
-                case exprAndHeaderFromText filePath contents of
-                    Right (header, _) -> return (absFile, header)
-                    _ -> Nothing
-
+        contents <- Text.IO.readFile filePath
+        case exprAndHeaderFromText filePath contents of
+            Right (header, _) -> return $ Just (absFile, removeComments header)
+            Left ParseError{..} -> do
+                putStrLn $ showParseError unwrap
+                return Nothing
+
+    removeComments :: Header -> Header
+    removeComments (Header header)
+        | "--" `Data.Text.isPrefixOf` strippedHeader = Header $ Data.Text.drop 2 strippedHeader
```

Comment:
Aside: I thought we wanted to distinguish between documentation comments and "internal" comments. Is this still planned?

Reply:
Although on the discourse [thread](https://discourse.dhall-lang.org/t/dhocs-cli-utility/267/24) for `dhall-docs` we didn't reach an agreement about it, I think it will be better to add the distintion.

Currently I don't because of a simple reason: The `prelude` comment headers are "internal" comments if we say so. I'd have to update prelude later. But for testing purposes, I'd prefer to add the distintion for "internal" and documentation comments later.

Reply:
Agree. Shall I ask again on the thread? Or add another thread?

Reply:
I'd be curious what @Gabriel439  thinks about this first. Continuing the discussion in the same thread seems fine to me though.

Reply:
We could do the same thing that Haskell does and use `{-|` or `-- |` to denote comments that should be interpreted by the documentation generator

Reply:
I'm going to update the prelude to have this change and what we have discussed at #1868 after dhall-lang/dhall-lang#1026 is merged. Meanwhile, I'll keep it as it currently is.